src/app/modules/convai-checker/perspectiveapi.service.ts

```diff
@@ -130,11 +130,42 @@ export class PerspectiveApiService {
         console.error('No server url specified for a non-direct API call.'
                       + ' Defaulting to current hosted address');
       }
-      const headers = new HttpHeaders();
-      headers.append('Content-Type', 'application/json');
 
-      return this.httpClient.post(
-        serverUrl + '/suggest_score', data, {headers});
+      return this
+        .sendRequest<SuggestCommentScoreData, SuggestCommentScoreResponse>(
+          data, `${serverUrl}/suggest_score`, 'suggestScore');
+    }
+  }
+
+  /**
+   * Issues a POST request to an endpoint.
+   *
+   * This fetches and attaches a reCAPTCHA token if RecaptchaV3Service is
+   * provided in the module.
+   *
+   * @param data The data to send in the POST request body.
+   * @param endpoint The endpoint to issue the PSOT request against.
+   * @param action The action name to attach to the reCATPCHA request for
+   * verification. This is used if the RecaptchaV3Service is provided. See
+   * https://developers.google.com/recaptcha/docs/v3#actions for more details.
+   */
+  private sendRequest<T, R>(data: T, endpoint: string, action: string):
+    Observable<R> {
+    if (this.recaptchaV3Service == null) {
```

Comment:
(Sorry, I can't seem to reply to the other thread). If it's definitely null, then you can just use ===. the == was in case it could be either.

Reply:
Ah! Done.